src/storage/store/engine.rs

```diff
@@ -113,6 +123,17 @@ impl BlockEngine {
         }
     }
 
+    pub fn get_last_block(&self) -> Option<Block> {
```

Comment:
should this return a result and push the decision to log, etc, to the caller?

Reply:
Can we make this change after we decide our error-handling type situation. If we want to use HubError, I'll change to that and expose HubError in all our interfaces.